### PR TITLE
Corpeventrecaptcha

### PR DESCRIPTION
**Pull request by Whamo12**
reCAPTCHA implementation for the corporate event  inquiry form.

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -131,33 +128,44 @@ router.post('/contactFormSubmit', function(req, res){
...
+        auth: {
+            user: config.gmail.user_name,
+            pass: config.gmail.password
+        }
+      })
+      var mailOptions = {
+        from: config.gmail.from,
+        to: config.gmail.mailTo,
+        subject: 'New Corporate Event Form Submission from ' + req.body.firstName + ' ' + req.body.lastName, // Subject line
+        text:  `From: ${req.body.firstName} ${req.body.lastName}
```
Comment (lojikil) [inline server/routes/api.js]: This actually sends this as plain text, ja? just making sure vis-a-vis SMTP Injection or HTML injection

File: src/app/app.service.spec.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ describe('AppService', () => {
       appService.getFilmGallery().subscribe(
         images => expect(images).toEqual(dummyImages, 'should return expected images')
     );
-      const req = httpMock.expectOne('http://localhost:3000/api/gallery_film');
+      const req = httpMock.expectOne('https://blooming-reef-68251.herokuapp.com/api/gallery_film');
```
Comment (lojikil) [inline src/app/app.service.spec.ts]: is this meant to have a hard-coded Heroku endpoint in here?

File: src/app/app.service.spec.ts [hunk 3]
```diff
@@ -53,7 +53,7 @@ describe('AppService', () => {
         images => expect(images.length).toEqual(0, 'should have empty images array')
       );
 
-      const req = httpMock.expectOne('http://localhost:3000/api/gallery_film');
+      const req = httpMock.expectOne('https://blooming-reef-68251.herokuapp.com/api/gallery_film');
```
Comment (lojikil) [inline src/app/app.service.spec.ts]: same as above vis-a-vis hard coding

File: src/app/contact/contact.component.spec.ts [hunk 4]
```diff
@@ -103,6 +106,18 @@ describe('ContactComponent', () => {
     expect(errors['required']).toBeTruthy();
   });
 
+  it('captcha validity', () => {
```
Comment (lojikil) [inline src/app/contact/contact.component.spec.ts]: <3 this

File: src/app/corporate-event/corporate-event.component.html [hunk 5]
```diff
@@ -60,6 +60,9 @@ <h2>Please complete the form below:</h2>
       <div class="form-group">
         <textarea formControlName="additionalInfo" type="textarea" class="form-control" id="additionalInfo" placeholder="Additional Info"></textarea>
       </div>
+      <div class="form-group">
+          <re-captcha formControlName="captcha" (resolved)="resolved($event)" siteKey="6LfpwmUUAAAAAN33yLrToNNR6leqqoIoRDTu0mWH" style="border-left: 0px"></re-captcha>
```
Comment (lojikil) [inline src/app/corporate-event/corporate-event.component.html]: can you extract the `siteKey` here a bit?

File: config.js [hunk 6]
```diff
@@ -2,14 +2,13 @@ var config = {}
 
 config.gmail = {}
 config.recaptcha = {}
+config.recaptcha.corpevent = {}
 
 config.gmail.user_name = process.env.GMAIL_USERNAME
```
Comment (lojikil) [inline config.js]: you may also want to move away from username & password towards the GMail OAuth API or the like. means not having to maintain those accounts, and just allowing the app to using the normal endpoints

File: src/app/app.service.spec.ts (hunk 2 above)
Comment (lojikil) [inline src/app/app.service.spec.ts]: did you mean to hardcode these Heroku endpoints in these?

File: src/app/corporate-event/corporate-event.component.html (hunk 5 above)
Comment (lojikil) [inline src/app/corporate-event/corporate-event.component.html]: can you extract the `siteKey` to be configurable from a config file? Also will help for others not to have your key, no?

File: src/app/contact/contact.component.spec.ts (hunk 4 above)
Comment (lojikil) [inline src/app/contact/contact.component.spec.ts]: ❤️ this

File: src/app/corporate-event/corporate-event.component.ts [hunk 7]
```diff
@@ -17,6 +17,10 @@ export class CorporateEventComponent implements OnChanges {
   alertType: String;
   alertMessage: String;
 
+  resolved(captchaResponse: string) {
+     // console.log(`Resolved captcha with response ${captchaResponse}:`);
```
Comment (lojikil) [inline src/app/corporate-event/corporate-event.component.ts]: why the dead logging code?

File: server/routes/api.js (hunk 1 above)
Comment (Whamo12) [reply server/routes/api.js]: Plain text.  I'll add extra validation.

**Pull request merged**